tests/e2e/run-local.sh

```diff
@@ -29,36 +30,47 @@ usage() {
                          the tests. Defaults to "kata-qemu".
 	-u: undo the installation and configuration before exiting. Useful for
 	    baremetal machine were it needs to do clean up for the next tests.
+	-t: enable default timeout for each operation (useful for CI)
 	EOF
 }
 
 parse_args() {
-	while getopts "hr:u" opt; do
+	while getopts "hr:ut" opt; do
 		case $opt in
 			h) usage && exit 0;;
 			r) runtimeclass="$OPTARG";;
 			u) undo="true";;
+			t) timeout="true";;
 			*) usage && exit 1;;
 		esac
 	done
 }
 
+run() {
+	duration=$1; shift
+	if [ "$timeout" == "true" ]; then
+		timeout $duration "$@"
```

Comment:
What if it prints a friendly message (e.g. "Run timed out after XX") when it timed out? i.e. when  `$?` -eq `124` ?

Reply:
That can open a can of worms as the script itself can return 124 so we'd have to add a logic to get the actual time (I mean we could use the $SECONDS so it's not that extensive but still) and then report "Run probably timed out after XXXs" when the timeout seems correct. Do you want me to add it or are we going to rely on log timestamps only?